### PR TITLE
feat: add UvDailyChart widget for daily UV index visualization

### LINKED ISSUE #20: feat: implement weekly UV bar chart widget
## Goal
Implement the 7-day UV bar chart as a standalone widget.

## Scope
- Create `lib/widgets/uv_daily_chart.dart`
- 7-day bar chart, left = current day, right = last day in forecast
- Each bar = daily UV max value
- Bars colored to WHO scale: 0-2 green, 3-5 yellow, 6-7 orange, 8-10 red, 11+ purple
- UV max value labeled on each bar
- Day labels abbreviated (Mon, Tue, Wed, etc.)
- Tight clinical bar spacing, no current-day highlight (position conveys it)
- Each bar navigable via TalkBack with semantic label: '[day], UV max [value], [risk] risk'
- Add widget tests for bar count, day labels, and WHO color mapping

## Out of scope
- Scrub or tap interactions
- Connecting to live `uvProvider` (done when wired into dashboard layout)

## References
- `.private/architecture/SCREENS.md` — Weekly Chart spec
- `docs/adr/0011-accessibility-approach.md`

### PR DESCRIPTION
**Pull request by milliorn**
- Introduce a new UvDailyChart widget to visualize the daily UV index over a week.

- Implement corresponding tests to ensure functionality and accuracy of the chart.

- Ensure the chart displays UV index values with appropriate color coding based on WHO risk levels.

Fixes #20

### CODE REVIEW

File: lib/widgets/uv_daily_chart.dart [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+              bottom: 0,
+              child: Semantics(
+                label: _pointLabel(point),
+                child: const SizedBox.expand(),
+              ),
+            ),
+        ],
+      ),
+    );
+  }
```
[the remark this answers is not in the document]
Comment (milliorn) [reply lib/widgets/uv_daily_chart.dart:336]: Confirmed and fixed. Verified against `uv_hourly_chart.dart`'s `_HourlyChartSemantics` (full-width `Expanded` regions) — the daily chart's nodes were indeed pinned to exactly `_barWidth` (20dp), much narrower.

Fix: each semantics node's cell now spans the full constant spacing between adjacent bar centers under `spaceEvenly` (applied symmetrically on both sides, including at the outermost bars), rather than just `_barWidth`. This keeps every node's rect exactly centered on its bar — verified via a widget test asserting `node.rect.center.dx` against the `_barCenterX` formula — while giving TalkBack a touch target roughly the width of a full day-cell instead of 20dp.

Also added a test for the single-entry case (`points.length == 1`, no neighbors on either side), which the fix's edge-case branch needed and wasn't previously covered.

File: lib/widgets/uv_daily_chart.dart [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+/// day (right).
+///
+/// Draws one WHO-colored bar per day with its UV max value labeled on top,
+/// and a day-of-week abbreviation on the bottom axis. Does not highlight
+/// the current day -- its leftmost position conveys that. Does not support
+/// tap/scrub interaction -- see the "Out of scope" note on the originating
+/// issue for that follow-up.
+class UvDailyChart extends StatelessWidget {
+  /// Creates a [UvDailyChart] from up to the first [_daysShown] entries in
+  /// [UvData.daily].
```
[the remark this answers is not in the document]
Comment (milliorn) [reply lib/widgets/uv_daily_chart.dart]: Confirmed and fixed. Updated both doc comments to mention the chronological sort and the stale-day (pre-today, location-local) filtering, so they match the actual implementation.

File: lib/widgets/uv_daily_chart.dart [hunk 3]
```diff
@@ -0,0 +1,317 @@
...
+  const UvDailyChart({required this.uvData, super.key});
+
+  /// The UV data to chart; only the first [_daysShown] entries of
+  /// [UvData.daily], sorted chronologically, are shown.
```
[the remark this answers is not in the document]
Comment (milliorn) [reply lib/widgets/uv_daily_chart.dart]: Confirmed and fixed alongside the constructor doc -- both now describe the sort + stale-day filter before the up-to-7 cap.

File: test/uv_daily_chart_test.dart [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+/// First day used by [makeUvData]'s defaults, so daily fixtures line up.
+///
+/// Anchored to today (UTC midnight) rather than a fixed historical date,
+/// because [UvDailyChart] now drops any daily entry whose location-local
+/// date is in the past -- a hardcoded past date would be silently filtered
+/// out of every fixture below.
+final DateTime _day0 = () {
+  final DateTime now = DateTime.now().toUtc();
+  return DateTime.utc(now.year, now.month, now.day);
+}();
```
[the remark this answers is not in the document]
Comment (milliorn) [reply test/uv_daily_chart_test.dart]: Confirmed and fixed. `_day0` is now a `late` variable assigned fresh in a `setUp()` block (computed the same way, just per-test instead of at file-load time), so it can never drift from the widget's own `DateTime.now()` call at build time regardless of how long the suite runs or when it crosses UTC midnight.

File: test/uv_daily_chart_test.dart [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+];
+
+String _weekdayAbbreviation(DateTime date) =>
+    _weekdayAbbreviations[date.weekday - 1];
+
+/// Abbreviated weekday names for [_day0] and the six days after it.
+final List<String> _weekdaysFromDay0 = <String>[
+  for (int i = 0; i < 7; i++)
+    _weekdayAbbreviation(_day0.add(Duration(days: i))),
+];
```
[the remark this answers is not in the document]
Comment (milliorn) [reply test/uv_daily_chart_test.dart]: Confirmed and fixed alongside `_day0` -- `_weekdaysFromDay0` is now also `late` and assigned in the same `setUp()` block, derived from that test run's `_day0` so labels can never desync from the fixtures.

File: test/uv_daily_chart_test.dart [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+  axisPosition: 0,
+  appliedInterval: 1,
+  sideTitles: const SideTitles(),
+  formattedValue: '',
+  axisSide: AxisSide.bottom,
+  rotationQuarterTurns: 0,
+);
+
+void main() {
+  testWidgets('renders one bar group per daily entry, up to 7', (
```
[the remark this answers is not in the document]
Comment (milliorn) [reply test/uv_daily_chart_test.dart:112]: Done -- added a `setUp()` at the top of `main()` that assigns both `_day0` and `_weekdaysFromDay0` from `DateTime.now().toUtc()` before each test. Verified: `flutter analyze --fatal-infos` clean, full suite (288 tests) passing, 100% line coverage maintained project-wide, `dartdoc` clean.

File: lib/widgets/uv_daily_chart.dart [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+  'Sat',
+  'Sun',
+];
+
+/// A single daily UV reading positioned along the chart's x-axis.
+///
+/// `index` is the bar's position (0 = current/earliest day); `localTime` is
+/// the reading's date in the queried location's local time (not the
+/// device's); `whoColor` is this bar's WHO risk-band color, computed once
+/// and reused by both the bar fill and the accessibility label.
```
[the remark this answers is not in the document]
Comment (milliorn) [reply lib/widgets/uv_daily_chart.dart]: Confirmed and fixed. `whoColor` is only read once, at the bar-fill `color:` -- the accessibility label derives its risk text independently via `whoRiskLabel(entry.uvi)`, never from `whoColor`. Updated the doc to say so.

[the remark this answers is not in the document]
Comment (milliorn) [reply lib/widgets/uv_daily_chart.dart]: Confirmed -- already fixed locally in a prior pass (uncommitted, not yet pushed, which is why this was re-flagged against the unfixed code still on the branch). The doc now reads: "`whoColor` is this bar's WHO risk-band color, computed once and used for the bar fill. The accessibility label derives its own risk text separately via `whoRiskLabel(entry.uvi)`, not from `whoColor`."

File: lib/widgets/uv_daily_chart.dart [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+    }
+    return (_barCenterX(1) - _barCenterX(0)) / 2;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Semantics(
+      container: true,
+      explicitChildNodes: true,
+      child: Stack(
```
[the remark this answers is not in the document]
Comment (milliorn) [reply lib/widgets/uv_daily_chart.dart:313]: Confirmed and fixed -- added a `constraints.maxWidth.isFinite` runtime guard in `UvDailyChart.build()`'s `LayoutBuilder`, returning `SizedBox.shrink()` when non-finite. Worth noting where I put it: an unbounded width actually crashes fl_chart's own `BarChart`/`SideTitlesWidget` first (confirmed by reproducing it with a real widget test placing the chart in a horizontal `SingleChildScrollView`), before `_DailyChartSemantics._barCenterX` is ever reached -- so the guard has to sit at the outer `UvDailyChart.build()` level, wrapping the whole `Stack`, not just inside `_DailyChartSemantics`. Deliberately did *not* also short-circuit on an empty `points` list here (unlike the assert this replaces) -- that case has no non-finite risk and was already rendering a valid empty chart; short-circuiting it too regressed that behavior, caught by the existing "daily is empty" test failing when I tried it.

Added a new test (`renders without crashing when laid out with unbounded width`) that reproduces the real crash via a horizontal `SingleChildScrollView` ancestor and asserts no exception escapes `pumpWidget`.

File: lib/widgets/uv_daily_chart.dart [hunk 9]
```diff
@@ -0,0 +1,320 @@
...
+                              top: Radius.circular(_barTopRadius),
+                            ),
+                            label: BarChartRodLabel(
+                              offset: const Offset(0, _valueLabelOffsetY),
+                              text: truncateToTenth(
+                                point.entry.uvi,
+                              ).toStringAsFixed(1),
+                              style: labelStyle,
+                            ),
+                          ),
```
[the remark this answers is not in the document]
Comment (milliorn) [reply lib/widgets/uv_daily_chart.dart:215]: Confirmed and fixed. Pinned `show: true` on `BarChartRodLabel`, matching the `alignment: BarChartAlignment.spaceEvenly` pin's existing rationale a few lines above. Note `show` has a literal default in fl_chart's `FlLabel` base class, so `avoid_redundant_argument_values` flags it as fatal under this repo's `--fatal-infos` gate -- added a targeted `// ignore:` comment for it, the first such suppression in `lib/`, since `alignment`'s pin avoids the lint only because its default is resolved via nullable-parameter fallback rather than a literal default value the analyzer can see.

File: test/uv_daily_chart_test.dart [hunk 10]
```diff
@@ -0,0 +1,470 @@
...
+        await tester.pumpWidget(_wrap(uvData));
+
+        expect(_chartData(tester).barGroups, hasLength(1));
+        final SemanticsNode node = tester.getSemantics(
+          find.bySemanticsLabel(RegExp('UV max')),
+        );
+        // With only one bar there is no neighbor on either side, so the
+        // node's cell falls back to half the plot width -- still centered
+        // on the single bar, not clipped to _barWidth.
+        expect(node.rect.width, greaterThan(barWidth));
```
[the remark this answers is not in the document]
Comment (milliorn) [reply test/uv_daily_chart_test.dart]: Confirmed and fixed. `barWidth` is now read from the rendered `BarChartData` (`data.barGroups.single.barRods.single.width`) instead of hardcoded, so this assertion can't silently desync from the widget's private `_barWidth` constant. Verified the test still passes with the derived value, confirming it matches what the hardcoded `20` used to be.

File: test/uv_daily_chart_test.dart [hunk 11]
```diff
@@ -0,0 +1,470 @@
...
+
+          // Mirrors fl_chart's BarChartAlignment.spaceEvenly formula
+          // (calculateGroupsX in bar_chart_data_extension.dart) so this
+          // test fails if the semantics overlay ever drifts from the
+          // chart's actual bar layout again.
+          const double spaceAvailable = chartWidth - barWidth * count;
+          const double eachSpace = spaceAvailable / (count + 1);
+          double expectedCenter(int index) =>
+              eachSpace * (index + 1) + barWidth * (index + 0.5);
+
```
[the remark this answers is not in the document]
Comment (milliorn) [reply test/uv_daily_chart_test.dart:438]: Confirmed and fixed the same way as the sibling single-entry test -- `barWidth` is now read from the rendered `BarChartData` (`_chartData(tester).barGroups.first.barRods.single.width`) before computing expected centers, instead of a hardcoded literal duplicating `_barWidth`.
